app/controllers/TermsAndConditionsController.scala

```diff
@@ -51,8 +54,14 @@ class TermsAndConditionsController @Inject() (
   val termsAndConditionsSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
     request.journey match {
       case j: Journey.BeforeConfirmedDirectDebitDetails => JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPageF(j)
-      case _: Journey.AfterConfirmedDirectDebitDetails =>
-        val isEmailAddressRequired = appConfig.emailJourneyEnabled
+      case j: Journey.AfterConfirmedDirectDebitDetails =>
+        val journeyEligibilityCheckResult: EligibilityCheckResult = j match {
+          case _: Stages.SubmittedArrangement =>
+            Errors.throwServerErrorException("Journey is in completed state, terms and conditions cannot be resubmitted.")
+          case j1: Journey.AfterEligibilityChecked => j1.eligibilityCheckResult
+        }
+
+        val isEmailAddressRequired = appConfig.emailJourneyEnabled && journeyEligibilityCheckResult.regimeDigitalCorrespondence.fold(false)(_.value)
```

Comment:
This is pretty much the main part of the change

Reply:
minor point but a shorter alternative to `.fold(false)(_.value)` would be `.exists(_.value)`

Reply:
Very good point, I'll update 😂